spec/omniauth/strategy_spec.rb

```diff
@@ -654,6 +654,13 @@ def make_env(path = '/auth/test', props = {})
         expect(strategy.env['rack.session']['omniauth.params']).to eq('foo' => 'bar')
       end
 
+      it 'sets omniauth.headers on the request phase' do
+        OmniAuth.config.mock_auth[:test] = {}
+
+        strategy.call(make_env('/auth/test', 'HTTP_FOO' => 'bar'))
```

Comment:
Well, this PR was just intended to keep header values, which in rack env are always prepended with `HTTP_`.